include/awkward/kernel-utils.h

```diff
@@ -99,4 +99,44 @@ extern "C" {
 
 }
 
+template <typename T>
+void awkward_ListArray_combinations_step(
+  T** tocarry,
+  int64_t* toindex,
+  int64_t* fromindex,
+  int64_t j,
+  int64_t stop,
+  int64_t n,
+  bool replacement) {
```

Comment:
Wait—we should avoid putting code in a .h file unless absolutely necessary. (There are some cases that are absolutely necessary.) I suppose this is here because it's templated and gets instantiated by awkward_ListArray_combinations and awkward_RegularArray_combinations (in different files, now). But, as a utility (not part of the public API to be guaranteed in future versions), it's never used without `T = int64_t`, so maybe just define one function as a utility.